### PR TITLE
Typecheck part of models

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## Changes

- add types and enabled type checking for part of models

### CODE REVIEW

File: lightly/models/batchnorm.py [hunk 1]
```diff
@@ -63,10 +65,11 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:
                 self.eps,
             ).view(N, C, H, W)
         else:
+            assert self.running_mean is not None and self.running_var is not None
```
Comment (IgorSusmelj) [inline lightly/models/batchnorm.py:68]: I would avoid having assert statements in time critical places like forward passes. That's why I added the ignore in the first place.

File: lightly/loss/memory_bank.py [hunk 2]
```diff
@@ -68,9 +68,9 @@ def _init_memory_bank(self, dim: int) -> None:
         # we could use register buffers like in the moco repo
         # https://github.com/facebookresearch/moco but we don't
         # want to pollute our checkpoints
-        self.bank = torch.randn(dim, self.size).type_as(self.bank) # type: ignore
-        self.bank = torch.nn.functional.normalize(self.bank, dim=0)
-        self.bank_ptr = torch.zeros(1).type_as(self.bank_ptr) # type: ignore
+        bank: Tensor = torch.randn(dim, self.size).type_as(self.bank)
```
Comment (IgorSusmelj) [inline lightly/loss/memory_bank.py:71]: Let's be consistent everywhere? I'd use `Tensor` or `torch.Tensor` everywhere.

File: lightly/models/modules/nn_memory_bank.py [hunk 3]
```diff
@@ -38,9 +39,16 @@ class NNMemoryBankModule(MemoryBankModule):
...
     def __init__(self, size: int = 2**16):
+        if size <= 0:
+            raise ValueError(f"Memory bank size must be positive, got {size}.")
         super(NNMemoryBankModule, self).__init__(size)
 
-    def forward(self, output: torch.Tensor, update: bool = False):
+    def forward(
+        self,
+        output: torch.Tensor,
+        labels: Optional[torch.Tensor] = None,
```
Comment (guarin) [inline lightly/models/modules/nn_memory_bank.py:49]: Why was the labels argument added?

File: lightly/models/batchnorm.py [hunk 4]
```diff
@@ -21,15 +25,18 @@ class SplitBatchNorm(nn.BatchNorm2d):
 
     """
 
-    def __init__(self, num_features, num_splits, **kw):
+    running_mean: torch.Tensor
+    running_var: torch.Tensor
```
Comment (guarin) [inline lightly/models/batchnorm.py:29]: I think this should be moved to the `__init__` method as those are not class variables.

**Review by guarin — APPROVED** [review]
Left one comment, otherwise LGTM! I'll let you resolve and merge.

One minor thing you could do is replace all `torch.Tensor` types with `Tensor` only. Then it would be more consistent with the rest of the codebase.

File: lightly/models/modules/nn_memory_bank.py (hunk 3 above)
Comment (philippmwirth) [reply lightly/models/modules/nn_memory_bank.py:49]: I have no idea. Will remove it again.

Comment (philippmwirth) [reply lightly/models/modules/nn_memory_bank.py]: It's because of mismatching signatures. I added a TODO instead.

File: lightly/models/batchnorm.py (hunk 4 above)
Comment (philippmwirth) [reply lightly/models/batchnorm.py]: Unfortunately it doesn't work if I move it to `__init__`. I have reverted to using type ignores for now.

Comment (guarin) [reply lightly/models/batchnorm.py]: This is weird, I tried it and it worked for me.
